### PR TITLE
Add tests for JxBrowserManager logic and add test comments

### PR DESCRIPTION
**Pull request by helin24**

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
       <property name="BorderFactoryClass" class="java.lang.String" />
     </properties>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="IntelliJ IDEA Community Edition" project-jdk-type="IDEA JDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_14" default="true" project-jdk-name="IntelliJ IDEA Community Edition IC-202.6109.22" project-jdk-type="IDEA JDK">
```
Comment (jacob314) [inline .idea/misc.xml:8]: revert this change.

File: src/io/flutter/jxbrowser/EmbeddedBrowser.java [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
 public class EmbeddedBrowser {
   public void openPanel(ContentManager contentManager, String tabName, String url) {
+    System.setProperty("jxbrowser.license.key", "6P83ACG409A8WGBSFIBUARBTNCLB4U9JHNK8FA3N59XH5Y7ZULRCP50NN2O9TRTT4CDC");
```
Comment (jacob314) [inline src/io/flutter/jxbrowser/EmbeddedBrowser.java:23]: revert this as well.

File: src/io/flutter/jxbrowser/JxBrowserManager.java [hunk 3]
```diff
@@ -39,12 +41,12 @@
 // the class path.
 public class JxBrowserManager {
   private static JxBrowserManager manager;
-  private static final String DOWNLOAD_PATH = PathManager.getPluginsPath() + File.separatorChar + "flutter-intellij" + File.separatorChar + "jxbrowser";
+  protected static final String DOWNLOAD_PATH = PathManager.getPluginsPath() + File.separatorChar + "flutter-intellij" + File.separatorChar + "jxbrowser";
```
Comment (jacob314) [inline src/io/flutter/jxbrowser/JxBrowserManager.java:44]: feel free to add a @VisibleForTesting annotation to these protected fields to indicate they are only there for testing.

File: src/io/flutter/jxbrowser/JxBrowserManager.java [hunk 4]
```diff
@@ -39,12 +41,12 @@
...
   private static final AtomicReference<JxBrowserStatus> status = new AtomicReference<>(JxBrowserStatus.NOT_INSTALLED);
   private static final Logger LOG = Logger.getInstance(JxBrowserManager.class);
   // We will be gating JxBrowser features until all of the features are landed.
   // To test JxBrowser, set this to true and also add license key to VM options (-Djxbrowser.license.key=<key>).
-  public static final boolean ENABLE_JX_BROWSER = false;
+  public static final boolean ENABLE_JX_BROWSER = true;
```
Comment (jacob314) [inline src/io/flutter/jxbrowser/JxBrowserManager.java:49]: did you mean to leave this on as part of the CL?

File: src/io/flutter/jxbrowser/JxBrowserManager.java [hunk 5]
```diff
@@ -56,6 +58,10 @@ public static JxBrowserManager getInstance() {
     return manager;
   }
 
+  protected static void resetForTest() {
```
Comment (jacob314) [inline src/io/flutter/jxbrowser/JxBrowserManager.java:61]: mark as @VisibleForTesting

File: src/io/flutter/utils/FileUtils.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    }
+    return true;
+  }
+
+  public static boolean fileExists(String path) {
+    final File file = new File(path);
+    return file.exists();
+  }
+
+  public static boolean deleteFileIfExists(String path) {
```
Comment (jacob314) [inline src/io/flutter/utils/FileUtils.java:28]: nit: maybe just call this `deleteFile` and indicate in the comment that you gracefully handle files that don't exist.

File: src/io/flutter/utils/FileUtils.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+package io.flutter.utils;
+
+import com.intellij.util.lang.UrlClassLoader;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class FileUtils {
+  public static boolean makeDirectoryIfNotExists(String path) {
```
Comment (jacob314) [inline src/io/flutter/utils/FileUtils.java:15]: nit: maybe just call this `makeDirectory` or `mkdirs` and indicate in the comment that you gracefully handle directories that already exist.

File: src/io/flutter/utils/FileUtils.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+  public static boolean deleteFileIfExists(String path) {
+    final File file = new File(path);
+    if (file.exists()) {
+      return file.delete();
+    }
+    return true;
+  }
+
+  public static boolean loadClassWithClassLoader(ClassLoader classLoader, String path) {
```
Comment (jacob314) [inline src/io/flutter/utils/FileUtils.java:36]: nit: just call 
this `loadClass` as it is clear from the classLoader argument that the class is loaded with the class loader.

File: src/io/flutter/jxbrowser/JxBrowserManager.java (hunk 4 above)
Comment (helin24) [reply src/io/flutter/jxbrowser/JxBrowserManager.java:49]: I thought I was adding these testing changes in a separate branch but accidentally committed them.
